### ISSUE TITLE
AttributeError: 'module' object has no attribute 'config'

### ISSUE THREAD

**Issue by KrisHammerberg** [SciTools/iris#1632]
When trying to import iris module the import chain breaks down at some point around the 'config' module.
I find this strange since the import call before the one that starts the traceback is "import iris.config". So presumably, this is imported without issue. 

EDIT: Moreover, if the 'iris.config' module is really unavailable in the scope of the last call [import iris.unit], why doesn't it fail at line 41 when that module imports iris.config?  

Full traceback:
import iris
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/__init__.py", line 107, in <module>
    import iris.cube
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/cube.py", line 38, in <module>
    import iris.analysis
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/analysis/__init__.py", line 59, in <module>
    from iris.analysis._area_weighted import AreaWeightedRegridder
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/analysis/_area_weighted.py", line 22, in <module>
    from iris.analysis._interpolation import get_xy_dim_coords, snapshot_grid
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/analysis/_interpolation.py", line 30, in <module>
    from iris.analysis.cartography import wrap_lons as wrap_circular_points
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/analysis/cartography.py", line 34, in <module>
    import iris.coords
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/coords.py", line 35, in <module>
    import iris.aux_factory
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/aux_factory.py", line 30, in <module>
    from iris._cube_coord_common import CFVariableMixin
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/_cube_coord_common.py", line 25, in <module>
    import iris.unit
  File "/usr/local/lib/python2.7/dist-packages/Iris-1.8.0_DEV-py2.7.egg/iris/unit.py", line 191, in <module>
    _lib_ud = iris.config.get_option(
AttributeError: 'module' object has no attribute 'config'

**Comment by KrisHammerberg** [comment]
Also I should note that I'm running Ubuntu 14.04 my install was working before a recent update and reboot.

**Comment by rhattersley** [comment]
Strange!! To help someone else with Ubuntu 14.04 reproduce this problem, how did you install iris?

I'd try checking what `iris.config` holds after line 41, and seeing what happens between there and line 191, e.g. via `python -m pdb your-script-that-imports-iris.py`.

**Comment by KrisHammerberg** [comment]
I almost certain that I used the recommended python setup.py install method. However, it is possible I used pip (I had been installing lots of odd utilities and building packages from source- being lazy and using a package manager might have been too tempting.)

I uninstalled iris and reinstalled it using setup.py install, just to be 100% sure. Problem is gone, but without any explanation... it kind of makes me wish I had tried debugging first, but only a little bit.
